### PR TITLE
Computeralgebra Rundbrief

### PR DESCRIPTION
**Pull request by HechtiDerLachs**
Some preparations for our article.

CC: @afkafkafk13 , @simonbrandhorst

### CODE REVIEW

File: src/AlgebraicGeometry/Schemes/CoveredSchemes/Morphisms/Methods.jl [hunk 1]
```diff
@@ -158,3 +158,50 @@ function Base.show(io::IO, ::MIME"text/plain", f::AbsCoveredSchemeMorphism)
...
+    V1 = domain(first(V1_candidates))
+    x = gens(OO(V1))
+    @show x
+    lift_x = [preimage(pullback(i1[V1]), f) for f in x]
+    @show lift_x
+    pb_x = pullback(i2[V2]).(lift_x)
+    @show pb_x
+    pb_x = pullback(j1[U]).(pb_x)
+    @show pb_x
+    morphism_dict[U] = ClosedEmbedding(SpecMor(U, V1, pb_x, check=false), pb_I2(V1), check=false)
```
Comment (afkafkafk13) [inline src/AlgebraicGeometry/Schemes/CoveredSchemes/Morphisms/Methods.jl]: please remove the "@show" statements

File: test/AlgebraicGeometry/Schemes/ProjectiveSchemes.jl [hunk 2]
```diff
@@ -312,3 +312,36 @@ end
...
+@testset "closed embeddings" begin
+  IP2 = projective_space(QQ, 2)
+  S = homogeneous_coordinate_ring(IP2)
+  (x, y, z) = gens(S)
+  I = ideal(S, [x^2 + y^2 + z^2])
+  X, inc = sub(IP2, I)
+  X, inc = sub(IP2, gens(I))
+
+  J = ideal(S, [x+y+z])
+  X2, inc2 = sub(IP2, J)
```
Comment (afkafkafk13) [inline test/AlgebraicGeometry/Schemes/ProjectiveSchemes.jl:325]: Line 321 is unnecessary and overwritten by line 322. The signature of line 321 is also tested in line 325

**Review by afkafkafk13 — COMMENTED** [review]
Overall design looks good, but tests are failing.

I left two comments for smaller problems, which are easy to solve.

File: src/AlgebraicGeometry/Schemes/ProjectiveSchemes/Morphisms/Attributes.jl [hunk 3]
```diff
@@ -158,3 +167,11 @@ function map_on_affine_cones(phi::ProjectiveSchemeMor{<:AbsProjectiveScheme{<:Sp
   return phi.map_on_affine_cones::SpecOpenMor
 end
 
+
+########################################################################
+# Spectial implementations for closed embeddings
```
Comment (simonbrandhorst) [inline src/AlgebraicGeometry/Schemes/ProjectiveSchemes/Morphisms/Attributes.jl]:
```suggestion
# Special implementations for closed embeddings
```

File: src/AlgebraicGeometry/Schemes/ProjectiveSchemes/Morphisms/Constructors.jl [hunk 4]
```diff
@@ -149,3 +149,33 @@ identity_map(P::AbsProjectiveScheme) = ProjectiveSchemeMor(P, P,
                                                         check=false
                                                        )
 
+function sub(P::AbsProjectiveScheme, I::Ideal)
+  @req base_ring(I) === homogeneous_coordinate_ring(P) "ideal must be defined in the homogeneous coordinate ring of the scheme"
+  inc = ProjectiveClosedEmbedding(P, I)
+  return domain(inc), inc
+end
```
Comment (simonbrandhorst) [inline src/AlgebraicGeometry/Schemes/ProjectiveSchemes/Morphisms/Constructors.jl]: for `subscheme` we set the ambient space for the domain. Perhaps we should do the same here?

**Comment by HechtiDerLachs** [comment]
Ready for merge. All required tests have passed for commit https://github.com/oscar-system/Oscar.jl/pull/2854/commits/64150f418e92c0b94763f557783d480a514f1cac. Let's save some energy here.

**Review by simonbrandhorst — APPROVED** [review]

File: test/AlgebraicGeometry/Schemes/ProjectiveSchemes.jl (hunk 2 above)
Comment (simonbrandhorst) [reply test/AlgebraicGeometry/Schemes/ProjectiveSchemes.jl:325]: Here the test for the signature is more explicit. So I would say it is okay as is.

**Review by afkafkafk13 — APPROVED** [review]

File: test/AlgebraicGeometry/Schemes/ProjectiveSchemes.jl (hunk 2 above)
Comment (afkafkafk13) [reply test/AlgebraicGeometry/Schemes/ProjectiveSchemes.jl:325]: fine with me

**Pull request merged**